src/Faithlife.OAuth/OAuthUtility-HmacSha256.cs

```diff
@@ -0,0 +1,244 @@
+			if (formParameters != null)
+				additionalParameters.AddRange(formParameters);
+			return CreateHmacSha256AuthorizationHeaderValue(uri, httpMethod, consumerSecret, temporarySecret, GetHmacSha256Parameters(consumerToken, nonceCreator ?? s_nonceCreator, systemTime ?? s_systemTime, additionalParameters));
+		}
+
+		/// <summary>
+		/// Create OAuth authentication header using the HMACSHA256 signature method.
+		/// </summary>
+		public static string CreateHmacSha256AuthorizationHeaderValue(Uri uri, string httpMethod, string consumerToken, string consumerSecret, string accessToken, string accessSecret, INonceCreator? nonceCreator = null, ISystemTime? systemTime = null)
+		{
+			return CreateHmacSha256AuthorizationHeaderValue(uri, httpMethod, consumerSecret, accessSecret, GetHmacSha256Parameters(consumerToken, nonceCreator ?? s_nonceCreator, systemTime ?? s_systemTime, new List<KeyValuePair<string, string>> { NewKeyValuePair(OAuthConstants.Token, accessToken) }));
+		}
+
+		/// <summary>
+		/// Create OAuth authentication header using the HMACSHA256 signature method. Accepts arbitrary number of form parameters.
+		/// </summary>
+		public static string CreateHmacSha256AuthorizationHeaderValue(Uri uri, IEnumerable<KeyValuePair<string, string>> formParameters, string httpMethod, string consumerToken, string consumerSecret, string accessToken, string accessSecret, INonceCreator? nonceCreator = null, ISystemTime? systemTime = null)
+		{
+			List<KeyValuePair<string, string>> additionalParameters = new List<KeyValuePair<string, string>> { NewKeyValuePair(OAuthConstants.Token, accessToken) };
+			if (formParameters != null)
+				additionalParameters.AddRange(formParameters);
+			return CreateHmacSha256AuthorizationHeaderValue(uri, httpMethod, consumerSecret, accessSecret, GetHmacSha256Parameters(consumerToken, nonceCreator ?? s_nonceCreator, systemTime ?? s_systemTime, additionalParameters));
+		}
+
+		internal static string CreateHmacSha256Signature(string signatureBase, string consumerSecret, string? tokenSecret)
+		{
+			using var hmacsha256 = new HMACSHA256(Encoding.UTF8.GetBytes(CreatePlainTextSignature(consumerSecret, tokenSecret)));
+			return Convert.ToBase64String(hmacsha256.ComputeHash(Encoding.UTF8.GetBytes(signatureBase)));
+		}
+
+		private static string CreateHmacSha256AuthorizationHeaderValue(Uri uri, string httpMethod, string consumerSecret, string? tokenSecret, ICollection<KeyValuePair<string, string>> parameters)
+		{
+			string signatureBase = CreateSignatureBase(uri, httpMethod, parameters, out var newUri);
+
+			parameters.Add(new KeyValuePair<string, string>(OAuthConstants.Signature, CreateHmacSha256Signature(signatureBase, consumerSecret, tokenSecret)));
+
+			return OAuthConstants.HeaderPrefix + " " + parameters
+				.Where(parameter => parameter.Key.StartsWithOrdinal(OAuthConstants.ParameterPrefix) && parameter.Value != null)
+				.Select(p => "{0}=\"{1}\"".FormatInvariant(p.Key, PercentEncode(p.Value)))
+				.Join(",");
+		}
```

Comment:
Seems like an exact duplication of https://github.com/Faithlife/FaithlifeOAuth/pull/3/files#diff-ad41888dfe9c00c72d737c029571963764aa9efd71c54af4218ced38c5937039R212-R222 except for the signature.

Let's parameterise the method on `OAuthSignatureMethods` and avoid duplication.